enhancements/update/distribute-secondary-metadata-as-container-image.md

```diff
@@ -0,0 +1,184 @@
+---
+title: cincinnati-graph-data-package-distribute-and-consume-in-a-container-image-repository
+authors:
+  - "@steveeJ"
+reviewers:
+  - @openshift/openshift-team-cincinnati-maintainers
+  - @openshift/openshift-team-cincinnati
+  - @openshift/team-cincinnati-operator
+approvers:
+  - TBD
+creation-date: 2020-05-07
+last-updated: 2020-05-07
+status: provision
+see-also:
+-
+replaces: []
+superseded-by: []
+---
+
+# cincinnati-graph-data: package, distribute, and consume in a container image repository
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [x] Graduation criteria
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Open Questions
+
+### Content trust
+Trust in the container image registry and in the container image content is possibly an issue if we an approach where Cincinnati uses a mutable tag, e.g. `latest` tag to receive the most recent content from the container registry.
+Assuming we're using Quay as an image registry, [which does not support Docker Content Trust](https://support.coreos.com/hc/en-us/articles/115000289114-Docker-Content-Trust-support).
+If we decide that to ensure the image integrity, which means do we have which do not further complicate the workflow?
+
+### Product name for the CPaaS integration
+Which product name should this image be released under?
+Related to this is that it's currently unclear if the Cincinnati image itself will be published under its own product name or now.
+Further note: the repository for the graph-data images will not be semantically versioned.
```

Comment:
Creating a new product is a big deal as we need to create many things for a product. As a engineering organization we need to track each products. Product management also need to be on board for any new product .So all practical purpose we should assume it would use the same product as Cincinnati image and Cincinnati-operator.

Reply:
If that works by the practical means of setting up the fully automated pipeline for the image creation I'm fine with any option of product handling here.

Reply:
These two things have completely different lifecycles and release gates. It could be difficult and counter-productive to meld them into one product.

Cincinnati and its operator will release when they have new features and/or bug fixes. Release gates include CI jobs and QE that ensure cincinnati itself works correctly.

The graph data will "release" based on OCP testing and releases that add or remove edges. Looking at the commit log for the repo, it seems to [average more than one change per day](https://github.com/openshift/cincinnati-graph-data/commits/master). At this point, each of those changes is implicitly a "release".

It would be confusing and awkward to release cincinnati itself once or more per day, just because the graph data is changing. Completely different stakeholders should be involved in releasing cincinnati itself vs OCP's graph data. I think this would need to be a separate product from cincinnati.

@rthallisey WDYT?

Reply:
Thanks for your thoughts on this @mhrivnak. I fully agree with the semantics you're describing. What I'm lacking is knowledge and experience about the implications of what it means to be a product, both practically as well es theoretically. I would appreciate any help on this matter as I want this enhancement to be realistic.

Reply:
Ryan Hartman would be a good person to discuss that with.

Reply:
Ya I agree with what your saying Michael.  Maybe this constitutes a new product or perhaps openshift should make this part of their release ritual?  Either way, I think this needs to be discussed in a wider forum to include some folks from the openshift release team.

Reply:
I was under the impression that even if multiple images are part of a single product in CPaaS but they can be released separately. So if this is not true then we need different product for this.  Also I do not see any advantage of releasing this with OpenShift's release payload  as this would be very dynamic in nature and will change everyday.

Reply:
@steveeJ Lets take this up with product management. We need to decide this asap.

Reply:
@katherinedube please chime in here if you can :pray:

Reply:
I wouldn't block this enhancement on the point of the graph data being it's own product because the pending result isn't a blocker for the downstream implementation.   Like I mentioned in [this thread](https://coreos.slack.com/archives/CEGKQ43CP/p1596030391458400?thread_ts=1595955435.428600&cid=CEGKQ43CP), you can choose what containers end up in an errata, so there may be a way to ship with "just graph data" and "just OSUS containers".  Therefore, I think it's a good idea to assume they will be the same product and the details can be sorted later.

Reply:
I agree with @rthallisey . The decision about single vs multiple products is orthogonal to this enhancement.

Reply:
Cincinnati graph data is definitely something that should be part of OpenShift Container Platform (it's only ever used within this context.) I don't want to bog down the enhancement with implementation details, but it's something that needs to be worked alongside the rest of the work around how best to implement this.